airflow/jobs.py

```diff
@@ -932,13 +929,19 @@ def _execute_task_instances(self,
         :type states: Tuple[State]
         :return: None
         """
-        # Get all the relevant task instances
+        # Get all the queued task instances from associated with scheduled
+        # DagRuns.
         TI = models.TaskInstance
         task_instances_to_examine = (
             session
             .query(TI)
             .filter(TI.dag_id.in_(simple_dag_bag.dag_ids))
             .filter(TI.state.in_(states))
+            .join(DagRun, and_(TI.dag_id == DagRun.dag_id,
+                               TI.execution_date == DagRun.execution_date,
+                               DagRun.state == State.RUNNING,
+                               DagRun.run_id.like(DagRun.ID_PREFIX + '%')))
```

Comment:
@aoen @plypaul Looking at this once again, this change is unnecessary and needs to be reverted (it has a bug as it is only looking for 'scheduled' dag_runs while some are manually created.

TaskInstances will be set to 'scheduled' if they meet the criteria to run, als the ones up for retry. No task_instance will be send to the executor in another state than 'scheduled'. Thus in an earlier patch I removed the 'upforretry' parameter from _execute_state_instances.